Debug-backend-tests.md

```diff
@@ -0,0 +1,268 @@
+
+Ran 1 test in 1 test class.
+All tests passed.
+```
+
+If a test fails, you'll always see its console output:
+
+```text
+[datastore] Sep 19, 2021 3:34:41 PM io.gapi.emulators.grpc.GrpcServer$3 operationComplete
+[datastore] INFO: Adding handler(s) to newly registered Channel.
+[datastore] Sep 19, 2021 3:34:41 PM io.gapi.emulators.netty.HttpVersionRoutingHandler channelRead
+[datastore] INFO: Detected HTTP/2 connection.
+Error 1
+HELLO THERE
+test_is_oppia_server_already_running_when_ports_closed (scripts.run_e2e_tests_test.RunE2ETestsTests) ... FAIL
+
+======================================================================
+FAIL: test_is_oppia_server_already_running_when_ports_closed (scripts.run_e2e_tests_test.RunE2ETestsTests)
+----------------------------------------------------------------------
+Traceback (most recent call last):
+  File "/oppia/scripts/run_e2e_tests_test.py", line 70, in test_is_oppia_server_already_running_when_ports_closed
+    raise AssertionError('hi')
+AssertionError: hi
+
+----------------------------------------------------------------------
+Ran 1 test in 1.078s
+
+FAILED (failures=1)
+Traceback (most recent call last):
+  File "/oppia/core/tests/gae_suite.py", line 126, in <module>
+    main()
+  File "/oppia/core/tests/gae_suite.py", line 122, in main
+    result.testsRun, len(result.errors), len(result.failures)))
+Exception: Test suite failed: 1 tests run, 0 errors, 1 failures.
+
+19:34:42 ERROR scripts.run_e2e_tests_test.RunE2ETestsTests.test_is_oppia_server_already_running_when_ports_closed: 21.1 secs
+Error 1
+HELLO THERE
+test_is_oppia_server_already_running_when_ports_closed (scripts.run_e2e_tests_test.RunE2ETestsTests) ... FAIL
+
+======================================================================
+FAIL: test_is_oppia_server_already_running_when_ports_closed (scripts.run_e2e_tests_test.RunE2ETestsTests)
+----------------------------------------------------------------------
+Traceback (most recent call last):
+  File "/oppia/scripts/run_e2e_tests_test.py", line 70, in test_is_oppia_server_already_running_when_ports_closed
+    raise AssertionError('hi')
+AssertionError: hi
+
+----------------------------------------------------------------------
+Ran 1 test in 1.078s
+
+FAILED (failures=1)
+Traceback (most recent call last):
+  File "/oppia/core/tests/gae_suite.py", line 126, in <module>
+    main()
+  File "/oppia/core/tests/gae_suite.py", line 122, in main
+    result.testsRun, len(result.errors), len(result.failures)))
+Exception: Test suite failed: 1 tests run, 0 errors, 1 failures.
+
+Stopping Redis Server(name="redis-server", pid=38032)...
+Stopping Cloud Datastore Emulator(name="python2.7", pid=38015)...
+
++------------------+
+| SUMMARY OF TESTS |
++------------------+
+...
+```
+
+Notice that the `HELLO THERE` output appears above the summary of tests. Also notice that it appears twice. This is a bug that will soon be fixed by [#13886](https://github.com/oppia/oppia/pull/13886).
```

Comment:
May want an issue to update this doc after it's fixed :)

Reply:
updated!